ui-v2/app/adapters/policy.js

```diff
@@ -0,0 +1,91 @@
+import Adapter, {
+  REQUEST_CREATE,
+  REQUEST_UPDATE,
+  DATACENTER_QUERY_PARAM as API_DATACENTER_KEY,
+} from './application';
+
+import { PRIMARY_KEY, SLUG_KEY } from 'consul-ui/models/policy';
+import { FOREIGN_KEY as DATACENTER_KEY } from 'consul-ui/models/dc';
+import { OK as HTTP_OK } from 'consul-ui/utils/http/status';
+import { PUT as HTTP_PUT } from 'consul-ui/utils/http/method';
+
+export default Adapter.extend({
+  // _ajaxRequest(options) {
+  //   const hash = Object.assign(
+  //     {},
+  //     options,
+  //     {
+  //       success: function(payload, textStatus, jqXHR) {
+  //         setTimeout(
+  //           () => {
+  //             return options.success(...arguments);
+  //           },
+  //           2000
+
+  //         );
+  //       }
+  //     }
+  //   );
+  //   $.ajax(hash);
+  // },
+  urlForQuery: function(query, modelName) {
+    return this.appendURL('acl/policies', [], this.cleanQuery(query));
+  },
+  urlForQueryRecord: function(query, modelName) {
+    if (typeof query.id === 'undefined') {
+      throw new Error('You must specify an id');
+    }
+    return this.appendURL('acl/policy', [query.id], this.cleanQuery(query));
+  },
+  urlForCreateRecord: function(modelName, snapshot) {
+    return this.appendURL('acl/policy', [], {
+      [API_DATACENTER_KEY]: snapshot.attr(DATACENTER_KEY),
+    });
+  },
+  urlForUpdateRecord: function(id, modelName, snapshot) {
+    return this.appendURL('acl/policy', [snapshot.attr(SLUG_KEY)], {
+      [API_DATACENTER_KEY]: snapshot.attr(DATACENTER_KEY),
+    });
+  },
+  urlForDeleteRecord: function(id, modelName, snapshot) {
+    return this.appendURL('acl/policy', [snapshot.attr(SLUG_KEY)], {
+      [API_DATACENTER_KEY]: snapshot.attr(DATACENTER_KEY),
+    });
+  },
+  urlForTranslateRecord: function(modelName, snapshot) {
+    return this.appendURL('acl/policy/translate', [], {});
+  },
+  dataForRequest: function(params) {
```

Comment:
Yeah for sure, this just follows the rest of the app and the old ACLs system.

`_super` calls this in RESTAdapter

https://github.com/emberjs/data/blob/e20597e8b1a2a6d7a4882ff91cdbe7481bee9415/addon/adapters/rest.js#L1261

> RESTAdapter#ajax has been deprecated please use. `methodForRequest`, `urlForRequest`, `headersForRequest` or `dataForRequest` instead.

https://github.com/emberjs/data/blob/e20597e8b1a2a6d7a4882ff91cdbe7481bee9415/addon/adapters/rest.js#L1230

🤷‍♂️ 

I started a refactor to move this around a bit to move some of this into the serializer instead, I did all the tests so I could do that with confidence - just not been able to do the moving yet.

As this PR isn't doing that, it would be the wrong time to do it in this PR. Once this is out, and it is a good time, I'll continue the work to move this around slightly.

I'm half waiting to move to ember/ember-data 3.4 to be honest, I plan on jigging around most of the ember-data layer once that's done, I'd probably say the ember-data code is weirdly the most involved code in the app 🤷‍♂️ 

Remember all of this:

> > RESTAdapter#ajax has been deprecated please use. `methodForRequest`, `urlForRequest`, `headersForRequest` or `dataForRequest` instead.

Is what they took out of ember-data in 3.whatever it was

Triple 🤷‍♂️

Reply:
Ohhh that's the feature flag they moved into an addon - odd that that duplicates serializer stuff? 🤷‍♂️ for sure haha